File: 卡码/代码随想录/1/03-哈希表/C++/01-有效字母异位词/LeetCode_0242.cpp
//242. 有效的字母异位词
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
	bool isAnagram(string s, string t) {
		int hash[26]={0};
		for(int i=0;i<s.length();i++) hash[s[i]-'a']++;
		for(int i=0;i<t.length();i++) hash[t[i]-'a']--;
		for(int i:hash)
			if(i!=0) return false;
		return true;
  }
};

File: 卡码/代码随想录/1/03-哈希表/Js/01-有效字母异位词/LeetCode_0242.js
//242. 有效的字母异位词
const isAnagram=(s,t)=>{
  let hash=new Array(26).fill(0);
  for(let ch of s)
    hash[ch.codePointAt(0)-'a'.codePointAt(0)]++;
  for(let ch of t)
    hash[ch.codePointAt(0)-'a'.codePointAt(0)]--;
  for(let num of hash)
    if(num!=0) return false;
  return true;
}